CD 79500 RN (#2030)
harness/developer-hub: release-notes/continuous-delivery.md

Harness deploys changes to Harness SaaS clusters on a progressive basis. This means that the features and fixes that these release notes describe may not be immediately available in your cluster. To identify the cluster that hosts your account, go to the **Account Overview** page. 
:::

## Latest - June 07, 2023, version 79500

```mdx-code-block
<Tabs>
  <TabItem value="What's new">
```

> - The variables defined at a pipeline level can be used with the SSH Command step, and can be referenced using `$VariableName` inside the script. This is applicable for secrets as well. (CDS-70189)

Author: ok, removed.

  
  Harness has introduced the following expressions to retrieve the current status of the strategy node during pipeline execution: 
  * `<+strategy.currentStatus>`: Retrieves the `currentStatus` of the strategy node with the maximum depth.
  * `<+strategy.node.strategyNodeIdentifier.currentStatus>`: Retrieves the `currentStatus` of the strategy node with a specific identifier, `strategyNodeIdentifier`.
  * `<+strategy.node.get("strategyNodeIdentifier").currentStatus>`: Retrieves the `currentStatus` of the strategy node with a specific identifier, `strategyNodeIdentifier`.
- If any entities referenced in a pipeline are updated, a warning now appears in Pipeline Studio saying that reconciliation is needed. Previously, this warning appeared only when you manually tried to reconcile. (CDS-69672)
- The Harness Approval step now supports scheduled automatic approvals. (CDS-69415)
  
  For more information, go to [Harness Approval steps in CD stages](https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages/).

> - Edit Git details for pipelines stored using Harness Git Experience. (CDS-69130)

Author: Yes, it is in this list too. So I added it here so that it is present in the latest.

- Send emails to non-Harness users. (CDS-58625, ZD-42496)
  
  To send emails to non-Harness users, you must configure your own SMTP server and enable the **Enable Emails to be sent to non-Harness Users** default setting. This setting is available at Account, Org, and Project levels.

  For more information on how to send emails to non-Harness users, go to [Email step reference](https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/email_step/).

  Harness Delegate version 79500 is required for this feature.

```mdx-code-block
  </TabItem>
  <TabItem value="Early access">
```

- Scale down the last successful stage environment created by using a Blue Green Deployment strategy. (CDS-68527)
  
  This functionality is behind a feature flag, `CDS_BG_STAGE_SCALE_DOWN_STEP_NG`. 

  This functionality helps you efficiently manage your resources. The scale down step can be configured within the same stage or different stage based on your requirement.

>   During scale down, the `HorizontalPodAutoscaler` and `PodDisruptionBudget` resources are removed, and the Deployments, StatefulSets, DaemonSets and Deployment Configs resources are scaled down. Make sure that the infrastructure definition of these resources and the Blue Green deployment are the same. This is necessary as Harness identifies resources from the release history, which is mapped to a release name. If you configure a different infrastructure definition, it might lead to scaling down important resources.

Author: ```suggestion
  During scale down, the `HorizontalPodAutoscaler` and `PodDisruptionBudget` resources are removed, and the Deployments, StatefulSets, DaemonSets and Deployment Configs resources are scaled down. Make sure that the infrastructure definition of these resources and the Blue Green service are the same. This is necessary as Harness identifies resources from the release history, which is mapped to a release name. If you configure a different infrastructure definition, it might lead to scaling down important resources.
```

- The expression, `<+artifacts.primary.identifier>` was not working properly for Google Cloud Storage deployments. (CDS-68993, ZD-44217)
  
  This issue is fixed. You can now see the identifier of the source selected as primary when using the expression `<+artifacts.primary.identifier>`. This functionality is behind the feature flag, `CDS_ARTIFACTS_PRIMARY_IDENTIFIER`.
- Fixed an issue where the SSH and WinRM rollback were not skipped even if there were no successful previous deployments. (CDS-68583)
- Fixed an issue where optional fields in a JIRA Update step were saved as key-value pairs. (CDS-58174)
- SSH pipelines with GCR artifacts ran without populating the required runtime inputs. (CDS-55689)
  
  Fixed this issue by adding proper validations to GCR artifacts used for SSH pipelines.
- The expressions corresponding to objects like list, maps, and so on were incorrectly converted to string type using the Java `String.valueOf` method resulting in incorrect formatting. (CDS-71619)
  
  This issue is fixed and the output values for expressions are returned as JSON objects.

### Harness Manager delegate fixed issues

> The fixed issues below are available with version 79500 and do not require a new delegate version. For Harness Delegate version-specific fixed issues, go to [Delegate release notes](/release-notes/delegate).

Author: ```suggestion
The fixed issues below are available with version 79503 and do not require a new delegate version. For Harness Delegate version-specific fixed issues, go to [Delegate release notes](/release-notes/delegate).
```

  
  This functionality is behind a feature flag, `CDS_BG_STAGE_SCALE_DOWN_STEP_NG`. 

  This functionality helps you efficiently manage your resources. The scale down step can be configured within the same stage or different stage based on your requirement.

  During scale down, the Horizontal Pod AutoScaler and Pod Disruption Budget resources are removed, and the Deployments, StatefulSets, DaemonSets and Deployment Configs resources are scaled down. Make sure that the infrastructure definition of these resources and the Blue Green deployment are the same. This is necessary as Harness identifies resources from the release history, which is mapped to a release name. If you configure a different infrastructure definition, it might lead to scaling down important resources.

  Harness Manager Delegate version 79500 is required for this feature.
  

```mdx-code-block
  </TabItem>
  <TabItem value="Fixed issues">
```

- Links to Org or Account level service or environment in a pipeline were redirecting to the Project level entities. (CDS-70607)
  
  This issue is fixed by adding correct links to Org and Account level services and environments in the pipeline Deployments page.

> - Unable to trigger pipelines using custom webhook triggers because the latest enhancement to custom webhook triggers requires a unique token for authentication. (CDS-70320, ZD-45022)
>   
>   This issue is fixed by introducing a new API for custom webhook triggers, `/v3` that contains the token required for authenticating webhook triggers. 

Reviewer: ```suggestion
- Unable to trigger pipelines using custom webhook triggers because the latest enhancement to custom webhook triggers requires a unique token for authentication. (CDS-70320, ZD-45022)
  
  This issue was caused by introducing a new API for custom webhook triggers, `/v3`, that generates a token for authenticating webhook triggers. You can see the token when you copy the trigger cURL command in Harness. This issues is fixed by allowing users to continue to use the previous API, `v2`, when the feature flag `SPG_DISABLE_CUSTOM_WEBHOOK_V3_URL` is enabled.
```
